force-app/main/default/lwc/formulaBuilder/formulaBuilder.js

```diff
@@ -0,0 +1,96 @@
+import { LightningElement, track } from 'lwc';
+import parseFormula from '@salesforce/apex/FormulaBuilderController.parseFormula';
+
+export default class FormulaBuilder extends LightningElement {
+
+    connectedCallback() {
+        let fields = [];
+        let functions = [];
+        let operators = [];
+
+        this.contextVariables.forEach(variable => {
+            if (variable.Fields.length > 0) {
+                let objectName = variable.Name;
+                variable.Fields.forEach(field => {
+                    let fieldValue = objectName + '.' + field;
+                    fields.push({ label: field, value: fieldValue });
+                })
+            } else {
+                let fieldValue = variable.Name;
+                fields.push({ label: variable.Name, value: fieldValue });
+            }
+        })
+
+        this.fields = fields;
+
+        this.supportedFunctions.forEach(func => {
+            let funcValue = func + '()';
+            functions.push({ label: func, value: funcValue });
+        })
+
+        this.functions = functions;
+        
+        this.supportedOperators.forEach(operator => {
+            operators.push({ label: operator, value: operator });
+        })
+
+        this.operators = operators;
+    }
+
+    
+    @track fields;
+    @track functions;
+    @track operators;
+    @track showSpinner = false;
+    @track showError = false;
+
+    @track formula = '';
+    @track parsedFormula = '';
+
+    @track supportedOperators = ['+', '-', '/', '*', '==', '!=', '>', '<', '>=', '<=', '<>'];
+    @track supportedFunctions = [
+        'AND', 'OR', 'NOT','XOR', 'IF', 'CASE', 'LEN',  'SUBSTRING','LEFT','RIGHT',
+        'ISBLANK','ISPICKVAL','CONVERTID', 'ABS','ROUND','CEIL','FLOOR','SQRT','ACOS',
+        'ASIN','ATAN','COS','SIN','TAN','COSH','SINH','TANH','EXP','LOG','LOG10','RINT',
+        'SIGNUM','INTEGER', 'POW','MAX','MIN','MOD', 'TEXT','DATETIME','DECIMAL','BOOLEAN',
+        'DATE', 'DAY','MONTH','YEAR','HOURS','MINUTES','SECONDS', 'ADDDAYS','ADDMONTHS',
+        'ADDYEARS','ADDHOURS','ADDMINUTES','ADDSECONDS'
+    ];
+    @track contextVariables = [
```

Comment:
these test values need to be taken out. they should be passed in for testing

Reply:
@alexed1 updated